pkg/cmd/server/origin/master.go

```diff
@@ -609,8 +609,8 @@ func (c *MasterConfig) RunBuildImageChangeTriggerController() {
 func (c *MasterConfig) RunDeploymentController() {
 	_, kclient := c.DeploymentControllerClients()
 	env := []api.EnvVar{
-		{Name: "KUBERNETES_MASTER", Value: c.MasterAddr},
-		{Name: "OPENSHIFT_MASTER", Value: c.MasterAddr},
+		{Name: "KUBERNETES_MASTER", Value: c.MasterPublicAddr},
```

Comment:
There are cases where the public address won't be accessible from within the cluster, so I don't think just switching to that is correct.